File: 백준/Silver/1735. 분수 합/분수 합.py
import sys 
input = sys.stdin.readline 
sys.setrecursionlimit(10**6)
def get_gcd(x, y):
	while x > 0:
		x, y = y%x, x
	
	return y

a, b = map(int, input().split())
c, d = map(int, input().split())
gcd=get_gcd(b, d)
gcm = b*d/gcd 
a = a*(gcm/b)
c = c*(gcm/d)
new_gcd = get_gcd(a+c, gcm)
print(int((a+c)/new_gcd), int(gcm/new_gcd))

File: 백준/Silver/1735. 분수 합/분수 합.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func getGcd(a, b int) int {
	for a > 0 {
		a, b = b%a, a
	}
	return b
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()
	var a, b, c, d int
	fmt.Fscanf(reader, "%d %d\n%d %d\n", &a, &b, &c, &d)
	gcd := getGcd(b, d)
	gcm := b * d / gcd
	a = a * (gcm / b)
	c = c * (gcm / d)
	newGcd := getGcd(a+c, gcm)
	fmt.Fprintf(writer, "%d %d\n", (a+c)/newGcd, gcm/newGcd)
}
